spring-boot-project/spring-boot-actuator/src/main/java/org/springframework/boot/actuate/endpoint/Sanitizer.java

```diff
@@ -215,9 +216,28 @@ private String sanitizeUri(String value) {
 		Matcher matcher = URI_USERINFO_PATTERN.matcher(value);
 		String password = matcher.matches() ? matcher.group(1) : null;
 		if (password != null) {
-			return StringUtils.replace(value, ":" + password + "@", ":" + SanitizableData.SANITIZED_VALUE + "@");
+			value = StringUtils.replace(value, ":" + password + "@", ":" + SanitizableData.SANITIZED_VALUE + "@");
 		}
-		return value;
+		return sanitizeUriQueryParams(value);
+	}
+
+	private String sanitizeUriQueryParams(String value) {
+		Matcher matcher = URI_QUERY_PARAM_PATTERN.matcher(value);
+		StringBuilder sanitizedUri = new StringBuilder();
+		while (matcher.find()) {
+			if (matcher.groupCount() != 2) {
```

Comment:
How can this `if` be triggered? None of the tests trigger it, and from the regex this shouldn't be possible. Maybe just remove it?

Reply:
Done.